### PR TITLE
MGMT-16649: Add Ignition Endpoint Headers to Host and Agent APIs

### PR DESCRIPTION
**Pull request by CrystalChun**
Adds `IgnitionEndpointHTTPHeaders` to the host and agent APIs. This allows adding sending additional headers when pulling the ignition from other ignition servers such as the hypershift ignition server. The unstructured string allows for flexibility in case headers need to be added or removed in the future.

## List all the issues related to this PR
https://issues.redhat.com/browse/MGMT-16649
- [x] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: api/v1beta1/agent_types.go [hunk 1]
```diff
@@ -194,6 +194,8 @@ type AgentSpec struct {
 	IgnitionConfigOverrides string `json:"ignitionConfigOverrides,omitempty"`
 	// IgnitionEndpointTokenReference references a secret containing an Authorization Bearer token to fetch the ignition from ignition_endpoint_url.
 	IgnitionEndpointTokenReference *IgnitionEndpointTokenReference `json:"ignitionEndpointTokenReference,omitempty"`
+	// IgnitionEndpointHTTPHeaders is a json formatted string of the additional HTTP headers when fetching the ignition.
+	IgnitionEndpointHTTPHeaders string `json:"ignition_endpoint_http_headers,omitempty"`
```
Comment (avishayt) [inline api/v1beta1/agent_types.go:198]: How about this?
```suggestion
	IgnitionEndpointHTTPHeaders map[string]string `json:"ignition_endpoint_http_headers,omitempty"`
```

File: swagger.yaml [hunk 2]
```diff
@@ -4899,6 +4899,11 @@ definitions:
         x-nullable: true
         type: string
         description: A string which will be used as Authorization Bearer token to fetch the ignition from ignition_endpoint_url.
+      ignition_endpoint_http_headers:
+        x-go-custom-tag: gorm:"type:text"
+        type: string
```
Comment (avishayt) [inline swagger.yaml:4904]: I suggest modeling this the same as the node_label property right below this one

Comment (CrystalChun) [reply swagger.yaml]: Thank you! I've updated it to resemble node_label, how does it look?

File: api/v1beta1/agent_types.go (hunk 1 above)
Comment (CrystalChun) [reply api/v1beta1/agent_types.go]: Modified as suggested, thank you!

File: api/v1beta1/agent_types.go [hunk 3]
```diff
@@ -194,6 +194,8 @@ type AgentSpec struct {
 	IgnitionConfigOverrides string `json:"ignitionConfigOverrides,omitempty"`
 	// IgnitionEndpointTokenReference references a secret containing an Authorization Bearer token to fetch the ignition from ignition_endpoint_url.
 	IgnitionEndpointTokenReference *IgnitionEndpointTokenReference `json:"ignitionEndpointTokenReference,omitempty"`
+	// IgnitionEndpointHTTPHeaders is a json formatted string of the additional HTTP headers when fetching the ignition.
+	IgnitionEndpointHTTPHeaders map[string]string `json:"ignition_endpoint_http_headers,omitempty"`
```
Comment (avishayt) [inline api/v1beta1/agent_types.go:198]:
```suggestion
	IgnitionEndpointHTTPHeaders map[string]string `json:"ignitionEndpointHttpHeaders,omitempty"`
```

**Review by avishayt — COMMENTED** [review]
Small change, otherwise looks good

File: api/v1beta1/agent_types.go (hunk 3 above)
Comment (CrystalChun) [reply api/v1beta1/agent_types.go]: Modified as suggested, thank you!

**Review by avishayt — APPROVED** [review]

**Comment by CrystalChun** [comment]
/test edge-subsystem-kubeapi-aws

Flaky tests

**Pull request merged**